src/scripts/OutSystems/OSUI/Patterns/OverflowMenuAPI.ts

```diff
@@ -0,0 +1,181 @@
+
+				_overflowMenuMap.delete(_overflowMenu.uniqueId);
+			},
+		});
+
+		return result;
+	}
+
+	/**
+	 * Function that will return the Map with all the OverflowMenu instances at the page
+	 *
+	 * @export
+	 * @return {*}  {Array<string>}
+	 */
+	export function GetAllOverflowMenus(): Array<string> {
+		return OSFramework.OSUI.Helper.MapOperation.ExportKeys(_overflowMenuMap);
+	}
+
+	/**
+	 * Function that gets the instance of OverflowMenu by a given Id.
+	 *
+	 * @export
+	 * @param {string} overflowMenuId
+	 * @return {*}  {OSFramework.OSUI.Patterns.OverflowMenu.IOverflowMenu}
+	 */
+	export function GetOverflowMenuById(overflowMenuId: string): OSFramework.OSUI.Patterns.OverflowMenu.IOverflowMenu {
+		return OSFramework.OSUI.Helper.MapOperation.FindInMap(
+			OSFramework.OSUI.GlobalEnum.PatternName.OverflowMenu,
+			overflowMenuId,
+			_overflowMenuMap
+		) as OSFramework.OSUI.Patterns.OverflowMenu.IOverflowMenu;
+	}
+
+	/**
+	 * Function that will initialize the pattern instance.
+	 *
+	 * @export
+	 * @param {string} overflowMenuId ID of the OverflowMenu that will be initialized.
+	 * @return {*}  {OSFramework.OSUI.Patterns.OverflowMenu.IOverflowMenu}
+	 */
+	export function Initialize(overflowMenuId: string): OSFramework.OSUI.Patterns.OverflowMenu.IOverflowMenu {
+		const _overflowMenu = GetOverflowMenuById(overflowMenuId);
+
+		_overflowMenu.build();
+
+		return _overflowMenu;
+	}
+
+	/**
+	 * Function to register a callback on this pattern
+	 *
+	 * @export
+	 * @param {string} overflowMenuId
+	 * @param {string} eventName
+	 * @param {OSFramework.OSUI.GlobalCallbacks.Generic} callback
+	 * @return {*}  {string}
+	 */
+	export function RegisterCallback(
+		overflowMenuId: string,
+		eventName: string,
+		callback: OSFramework.OSUI.GlobalCallbacks.Generic
+	): string {
+		const result = OutSystems.OSUI.Utils.CreateApiResponse({
+			errorCode: ErrorCodes.OverflowMenu.FailRegisterCallback,
+			callback: () => {
+				const _overflowMenu = GetOverflowMenuById(overflowMenuId);
+
+				_overflowMenu.registerCallback(eventName, callback);
+			},
+		});
+
+		return result;
+	}
+
+	/**
+	 * Function to open this pattern
+	 *
+	 * @export
+	 * @param {string} overflowMenuId
+	 * @return {*}  {string}
+	 */
+	export function Open(overflowMenuId: string): string {
+		const result = OutSystems.OSUI.Utils.CreateApiResponse({
+			errorCode: ErrorCodes.OverflowMenu.FailOpen,
+			callback: () => {
+				const _overflowMenuItem = GetOverflowMenuById(overflowMenuId);
+
+				_overflowMenuItem.open(true);
+			},
+		});
+
+		return result;
+	}
+
+	/**
+	 * Function to close this pattern
+	 *
+	 * @export
+	 * @param {string} overflowMenuId
+	 * @return {*}  {string}
+	 */
+	export function Close(overflowMenuId: string): string {
+		const result = OutSystems.OSUI.Utils.CreateApiResponse({
+			errorCode: ErrorCodes.OverflowMenu.FailClose,
+			callback: () => {
+				const _overflowMenuItem = GetOverflowMenuById(overflowMenuId);
+
+				_overflowMenuItem.close();
```

Comment:
@BenOsodrac Couldn't we check if its actually opened before closing it?

Reply:
We do that on the Pattern class, not on the API